src/libs/ReportActionsUtils.ts

```diff
@@ -565,134 +486,125 @@ function getMostRecentReportActionLastModified() {
 }
 
 /**
  * A helper method to identify if the message is deleted or not.
- *
- * @param {Object} reportAction
- * @returns {Boolean}
  */
-function isMessageDeleted(reportAction) {
-    return lodashGet(reportAction, ['message', 0, 'isDeletedParentAction'], false);
+function isMessageDeleted(reportAction: OnyxEntry<ReportAction>): boolean {
+    return reportAction?.message?.[0]?.isDeletedParentAction ?? false;
 }
 
 /**
  * Returns the number of money requests associated with a report preview
- *
- * @param {Object|null} reportPreviewAction
- * @returns {Number}
  */
-function getNumberOfMoneyRequests(reportPreviewAction) {
-    return lodashGet(reportPreviewAction, 'childMoneyRequestCount', 0);
+function getNumberOfMoneyRequests(reportPreviewAction: OnyxEntry<ReportAction>): number {
+    return reportPreviewAction?.childMoneyRequestCount ?? 0;
 }
 
-/**
- * @param {*} reportAction
- * @returns {Boolean}
- */
-function isSplitBillAction(reportAction) {
-    return lodashGet(reportAction, 'originalMessage.type', '') === CONST.IOU.REPORT_ACTION_TYPE.SPLIT;
+function isSplitBillAction(reportAction: OnyxEntry<ReportAction>): boolean {
+    return reportAction?.actionName === CONST.REPORT.ACTIONS.TYPE.IOU && reportAction.originalMessage.type === CONST.IOU.REPORT_ACTION_TYPE.SPLIT;
 }
 
-/**
- *
- * @param {*} reportAction
- * @returns {Boolean}
- */
-function isTaskAction(reportAction) {
-    const reportActionName = lodashGet(reportAction, 'actionName', '');
+function isTaskAction(reportAction: OnyxEntry<ReportAction>): boolean {
+    const reportActionName = reportAction?.actionName;
     return (
         reportActionName === CONST.REPORT.ACTIONS.TYPE.TASKCOMPLETED ||
         reportActionName === CONST.REPORT.ACTIONS.TYPE.TASKCANCELLED ||
         reportActionName === CONST.REPORT.ACTIONS.TYPE.TASKREOPENED
     );
 }
 
-/**
- * @param {*} reportID
- * @returns {[Object]}
- */
-function getAllReportActions(reportID) {
-    return lodashGet(allReportActions, reportID, []);
+function getAllReportActions(reportID: string): ReportActions {
+    return allReportActions?.[reportID] ?? {};
 }
 
 /**
  * Check whether a report action is an attachment (a file, such as an image or a zip).
  *
- * @param {Object} reportAction report action
- * @returns {Boolean}
+ * @param reportAction report action
  */
-function isReportActionAttachment(reportAction) {
-    const message = _.first(lodashGet(reportAction, 'message', [{}]));
-    return _.has(reportAction, 'isAttachment') ? reportAction.isAttachment : isReportMessageAttachment(message);
+function isReportActionAttachment(reportAction: OnyxEntry<ReportAction>): boolean {
+    const message = reportAction?.message?.[0];
+
+    if (reportAction && 'isAttachment' in reportAction) {
+        return reportAction.isAttachment ?? false;
+    }
+
+    if (message) {
+        return isReportMessageAttachment(message) ?? false;
```

Comment:
NAB: I don't think we need `?? false` if `isReportMEssageAttachment` returns a boolean already.

Reply:
Resolved